usaspending_api/reporting/v2/views/agencies/overview.py

```diff
@@ -156,8 +156,13 @@ def get_agency_overview(self):
                 "unlinked_contract_award_count",
                 "unlinked_assistance_award_count",
             )
-            .order_by(f"{'-' if self.pagination.sort_order == 'desc' else ''}{self.pagination.sort_key}")
         )
+
+        if self.pagination.sort_order == "desc":
+            result_list = result_list.order_by(F(self.pagination.sort_key).desc(nulls_last=True))
```

Comment:
Would recommend COALESCE the `null` values to 0 instead of simply having them be first or last. In column where values can be negative it doesn't make sense. Screenshot of my local on your branch below:

![Screen Shot 2021-02-25 at 7 23 35 AM](https://user-images.githubusercontent.com/40359517/109175173-61f22600-773a-11eb-991c-4d23dae1676e.png)

Reply:
Would also recommend adding a secondary sort of `toptier_code` (or just some unique column that is non-null) to make sure we get a consistent sort.

Reply:
> 
> 
> Would recommend COALESCE the `null` values to 0 instead of simply having them be first or last. In column where values can be negative it doesn't make sense. Screenshot of my local on your branch below:
> 
Makes sense. I tried for over a day to get a COALESCE in here, failed to, and did what Tony suggested. But we hadn't thought of negatives.

Reply:
Technically speaking your columns in this ticket won't be displaying negatives, but because we hand't thought about it before now your changes will affect those columns.

Reply:
@sethstoudenmier My confusion: this is sorting amounts; my update only had to do with counts. What's the correlation here? Or is this a different update we need to fix?

Reply:
> By default, null values sort as if larger than any non-null value; that is, NULLS FIRST is the default for DESC order, and NULLS LAST otherwise.
(https://www.postgresql.org/docs/10/queries-order.html)

Before your changes the `NULL` values for the spending columns were sorting according to the description above. Now they are sorting out of order in a different way. We can bring up the option to create a different ticket to correct sort for those, but it is going to require undoing your changes to write up a new way to sort that correctly handles nulls for all columns.

Reply:
I think simply placing nulls last for both directions will fix the issue you reported @sethstoudenmier

Reply:
@sethstoudenmier I get it, now I'm resorting nulls in all values, not just the ones I'm fixing.

Reply:
@brett-varney your code changes are affecting to sort behavior for every field

> @sethstoudenmier My confusion: this is sorting amounts; my update only had to do with counts. What's the correlation here? Or is this a different update we need to fix?

@brett-varney your code changes affect sorting across all of the fields